packages/testcontainers/src/containers/NativeChainContainer.ts

```diff
@@ -0,0 +1,671 @@
+ */
+export class NativeChainContainer extends GenericContainer {
+  /**
+   * @param {string} image docker image name
+   */
+  constructor (
+    image: string = NativeChainContainer.image
+  ) {
+    super(image)
+  }
+
+  static get image (): string {
+    if (process?.env?.DEFICHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.DEFICHAIN_DOCKER_IMAGE
+    }
+    return 'defi/defichain:HEAD-02ef6a1b3'
+  }
+
+  public static readonly PREFIX = 'defichain-testcontainers-'
+
+  /**
+   * Generate a name for a new docker container with network type and random number
+   */
+  generateName (): string {
+    const rand = Math.floor(Math.random() * 10000000)
+    return `${NativeChainContainer.PREFIX}-${this.blockchainNetwork.name}-${rand}`
+  }
+
+  protected blockchainNetwork: BlockchainNetwork = getNetwork('testnet')
+
+  /**
+   * @param networkName Blockchain network name (e.g. 'testnet')
+   * @returns this
+   */
+  public withBlockchainNetwork (networkName: NetworkName): this {
+    this.blockchainNetwork = getNetwork(networkName)
+    return this
+  }
+
+  public static readonly DefaultRpcUser = 'testcontainers-user'
+  public static readonly DefaultRpcPassword = 'testcontainers-password'
+
+  protected rpcUser: string = NativeChainContainer.DefaultRpcUser
+  protected rpcPassword: string = NativeChainContainer.DefaultRpcPassword
+
+  public withRpcUser (rpcUser: string): this {
+    this.rpcUser = rpcUser
+    return this
+  }
+
+  public withRpcPassword (rpcPassword: string): this {
+    this.rpcPassword = rpcPassword
+    return this
+  }
+
+  protected masterNodeKey?: MasterNodeKey
+
+  public withMasterNodeKey (key: MasterNodeKey): this {
+    this.masterNodeKey = key
+    return this
+  }
+
+  protected hasPreconfiguredMasternode: boolean = false
+
+  public withPreconfiguredRegtestMasternode (masterNodeKey = RegTestFoundationKeys[0]): this {
+    this.withMasterNodeKey(masterNodeKey)
+    this.withBlockchainNetwork('regtest')
+    this.hasPreconfiguredMasternode = true
+    return this
+  }
+
+  public static readonly defaultCmd = {
+    prepend: [
+      'defid',
+      '-printtoconsole',
+      '-rpcallowip=0.0.0.0/0',
+      '-rpcbind=0.0.0.0',
+      '-rpcworkqueue=512'
+    ],
+    testnet: [
+      '-testnet=1'
+    ],
+    regtest: [
+      '-regtest=1',
+      '-jellyfish_regtest=1',
+      '-txnotokens=0',
+      '-logtimemicros',
+      '-txindex=1',
+      '-acindex=1',
+      '-amkheight=0',
+      '-bayfrontheight=1',
+      '-bayfrontgardensheight=2',
+      '-clarkequayheight=3',
+      '-dakotaheight=4',
+      '-dakotacrescentheight=5',
+      '-eunosheight=6',
+      '-eunospayaheight=7',
+      '-fortcanningheight=8',
+      '-fortcanningmuseumheight=9',
+      '-fortcanninghillheight=10',
+      '-fortcanningroadheight=11',
+      '-fortcanningcrunchheight=12',
+      '-fortcanningspringheight=13',
+      '-fortcanninggreatworldheight=14',
+      '-fortcanningepilogueheight=15'
+    ],
```

Comment:
I think there are missing grand central flags
[reference](https://github.com/JellyfishSDK/jellyfish/blob/43cd88ab5d78caf0c4627079f04a15769f48b530/packages/testcontainers/src/containers/RegTestContainer/index.ts#L43)

```
'-grandcentralheight=16',
'-regtest-skip-loan-collateral-validation',
'-regtest-minttoken-simulate-mainnet=0'
```

Reply:
About your question regarding both classes in the same file, I was following the implementation here: https://github.com/testcontainers/testcontainers-node/blob/6f8971796f5a9c50e9cc38651a87ba522a093470/src/modules/postgresql/postgresql-container.ts.

Not a very good answer, but I don't see any problems with both classes being in the same file, hence, no reason to deviate from their standard implementation.

Reply:
> About your question regarding both classes in the same file.

Yeah, I guess conventions. Contextually speaking, probably no point separating them into different files either since the context is the same.